### ISSUE TITLE
SRN 使用中文数据集训练，跑了快2个epoch了，acc 还是0, 是配置出了问题吗？

### ISSUE THREAD

**Issue by zsshub** [PaddlePaddle/PaddleOCR#2540]
[2021/04/17 19:06:17] root INFO: epoch: [2/72], iter: 9275, lr: 0.000100, loss: 4572.994141, word_loss: 1105.126587, img_loss: 1081.505615, acc: 0.000000, norm_edit_dis: 0.071429, reader_cost: 0.00008 s, batch_cost: 0.30322 s, samples: 160, ips: 105.53316
[2021/04/17 19:06:18] root INFO: epoch: [2/72], iter: 9280, lr: 0.000100, loss: 4620.915527, word_loss: 1115.209229, img_loss: 1098.085571, acc: 0.000000, norm_edit_dis: 0.079150, reader_cost: 0.00008 s, batch_cost: 0.30261 s, samples: 160, ips: 105.74663
[2021/04/17 19:06:20] root INFO: epoch: [2/72], iter: 9285, lr: 0.000100, loss: 4683.862305, word_loss: 1130.927368, img_loss: 1114.635254, acc: 0.000000, norm_edit_dis: 0.080007, reader_cost: 0.00008 s, batch_cost: 0.30439 s, samples: 160, ips: 105.12765
[2021/04/17 19:06:22] root INFO: epoch: [2/72], iter: 9290, lr: 0.000100, loss: 4392.288086, word_loss: 1058.817871, img_loss: 1049.120117, acc: 0.015625, norm_edit_dis: 0.076348, reader_cost: 0.00008 s, batch_cost: 0.30282 s, samples: 160, ips: 105.67248

### **rec_r50_fpn_srn.yml配置：**

Global:
  use_gpu: True
  epoch_num: 72
  log_smooth_window: 20
  print_batch_step: 5
  save_model_dir: ./output/rec/srn_new
  save_epoch_step: 3

  eval_batch_step: [0, 5000]

  cal_metric_during_train: True
  pretrained_model: 
  checkpoints:
  save_inference_dir:
  use_visualdl: False
  infer_img: doc/imgs_words/ch/word_1.jpg

  character_dict_path: ppocr/utils/ppocr_keys_v1.txt
  character_type: ch
  max_text_length: 50
  num_heads: 8
  infer_mode: False
  use_space_char: True
  distort: true

Optimizer:
  name: Adam
  beta1: 0.9
  beta2: 0.999
  clip_norm: 10.0
  lr:
    learning_rate: 0.001

Architecture:
  model_type: rec
  algorithm: SRN
  in_channels: 1
  Transform:
  Backbone:
    name: ResNetFPN
  Head:
    name: SRNHead
    max_text_length: 50
    num_heads: 8
    num_encoder_TUs: 2
    num_decoder_TUs: 4
    hidden_dims: 512

Loss:
  name: SRNLoss

PostProcess:
  name: SRNLabelDecode

Metric:
  name: RecMetric
  main_indicator: acc

Train:
  dataset:
    name: SimpleDataSet
    data_dir: /media/ime/FCC88331C882E8EA/OCR数据集/中文数据集/train_data/reg/train/
    label_file_list: ["/media/ime/FCC88331C882E8EA/OCR数据集/中文数据集/train_data/reg/rec_gt_train.txt"]
    transforms:
      - DecodeImage: # load image
          img_mode: BGR
          channel_first: False
      - SRNLabelEncode: # Class handling label
      - SRNRecResizeImg:
          image_shape: [3, 48, 320]
      - KeepKeys:
          keep_keys: ['image',
                      'label',
                      'length',
                      'encoder_word_pos',
                      'gsrm_word_pos',
                      'gsrm_slf_attn_bias1',
                      'gsrm_slf_attn_bias2'] # dataloader will return list in this order
  loader:
    shuffle: False
    batch_size_per_card: 32
    drop_last: False
    num_workers: 4

Eval:
  dataset:
    name: SimpleDataSet
    data_dir: /media/ime/FCC88331C882E8EA/OCR数据集/中文数据集/train_data/reg/test/
    label_file_list: ["/media/ime/FCC88331C882E8EA/OCR数据集/中文数据集/train_data/reg/rec_gt_test.txt"]
    transforms:
      - DecodeImage: # load image
          img_mode: BGR
          channel_first: False
      - SRNLabelEncode: # Class handling label
      - SRNRecResizeImg:
          image_shape: [3, 48, 320]
      - KeepKeys:
          keep_keys: ['image',
                      'label',
                      'length',
                      'encoder_word_pos',
                      'gsrm_word_pos',
                      'gsrm_slf_attn_bias1',
                      'gsrm_slf_attn_bias2'] 
  loader:
    shuffle: False
    drop_last: False
    batch_size_per_card: 32
    num_workers: 4

**Comment by cxy86121-sudo** [comment]
请问用SRNet 训练的是什么模型？

**Comment by MissPenguin** [comment]
max_text_length先不要改哈，代码里按照25写死了，改成50的话训练会有问题，这个我们后续优化下

**Comment by zsshub** [comment]
> 请问用SRNet 训练的是什么模型？
识别模型，数据集是中文街景文字识别，数据来源：https://aistudio.baidu.com/aistudio/competition/detail/8

**Comment by zsshub** [comment]
> max_text_length先不要改哈，代码里按照25写死了，改成50的话训练会有问题，这个我们后续优化下

请问可能会出现什么问题，我现在的模型收敛速度**很慢**跟这有关吗？

**Comment by zsshub** [comment]
> max_text_length先不要改哈，代码里按照25写死了，改成50的话训练会有问题，这个我们后续优化下

配置里image_shape: [3, 48, 320], 可以修改吗，原先是 [3, 32, 320] ？

**Comment by MissPenguin** [comment]
> > max_text_length先不要改哈，代码里按照25写死了，改成50的话训练会有问题，这个我们后续优化下
> 
> 请问可能会出现什么问题，我现在的模型收敛速度**很慢**跟这有关吗？

是的

**Comment by MissPenguin** [comment]
> > max_text_length先不要改哈，代码里按照25写死了，改成50的话训练会有问题，这个我们后续优化下
> 
> 配置里image_shape: [3, 48, 320], 可以修改吗，原先是 [3, 32, 320] ？

都先别改吧，你用原始配置训练看看先

**Comment by zsshub** [comment]
使用SRN训练中文数据集，配置文件里只需修改这两项就行了吗？
character_dict_path: ppocr/utils/ppocr_keys_v1.txt
character_type: ch

**Comment by zsshub** [comment]
使用SRN训练中文数据集，配置文件里只需修改这两项就行了吗？
character_dict_path: ppocr/utils/ppocr_keys_v1.txt
character_type: ch

**Comment by MissPenguin** [comment]
> 使用SRN训练中文数据集，配置文件里只需修改这两项就行了吗？
> character_dict_path: ppocr/utils/ppocr_keys_v1.txt
> character_type: ch

是的，另外use_space_char可以改成True

**Comment by chenyixun1112** [comment]
> > > max_text_length先不要改哈，代码里按照25写死了，改成50的话训练会有问题，这个我们后续优化下
> > 
> > 
> > 配置里image_shape: [3, 48, 320], 可以修改吗，原先是 [3, 32, 320] ？
> 
> 都先别改吧，你用原始配置训练看看先

请问这个问题优化了吗？
